src/Network.cpp

```diff
@@ -704,26 +702,31 @@ void batchnorm(const size_t channels,
                const float* const stddivs,
                const float* const eltwise = nullptr)
 {
-    const auto lambda_ReLU = [](auto val) { return (val > 0.0f) ?
-                                            val : 0.0f; };
-
-    for (auto c = size_t{0}; c < channels; ++c) {
-        const auto mean = means[c];
-        const auto scale_stddiv = stddivs[c];
-
-        if (eltwise == nullptr) {
-            // Classical BN
-            auto arr = &data[c * spatial_size];
+    if (eltwise == nullptr) {
+        // Classical BN
+        const auto lambda_ReLU = [](auto stddiv, auto val) { return (val > 0.0f) ?
```

Comment:
This isn't obviously an optimization to me, because it removes the possibility to vectorize the stddiv multiplication.